pip_audit/_dependency_source/resolvelib/pypi_provider.py

```diff
@@ -95,10 +101,15 @@ def metadata(self) -> Message:
         if self._metadata is None:
             self._state.update_state(f"Fetching metadata for {self.name} ({self.version})")
 
+            response = self._session.get(self.url, timeout=self._timeout)
```

Comment:
I moved this out here so that we can mock out the `get_metadata_for_X` functions to short-circuit dependency resolution without losing the hashes that are generated in `generate_dist_hashes`. It also removes a bit of redundancy.

Reply:
Makes sense!